resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -259,12 +259,19 @@ private[spark] object KubernetesUtils extends Logging {
       isLocalDependency(Utils.resolveURI(resource))
   }
 
-  def renameMainAppResource(resource: String, conf: SparkConf): String = {
+  def renameMainAppResource(
+      resource: String,
+      conf: Option[SparkConf] = None,
+      isJava: Boolean = true): String = {
     if (isLocalAndResolvable(resource)) {
-      SparkLauncher.NO_RESOURCE
+      if (isJava) {
```

Comment:
I think I'm missing some context, and others may be as well. Could we add a comment about why we need to upload the non-Java main resource?

Reply:
I upload it because it resided at the client machine

Reply:
So I guess the part I don't understand is why would it not need to be uploaded if it was Java? Presumably we upload the java somewhere else, why isn't the Python being uploaded in the same place?

Reply:
@holdenk does the java loading explaination work for you:
https://github.com/apache/spark/pull/25870#issuecomment-541001902

To confirm - the timing of the java loading is to ensure java code can execute when its needed?

Reply:
I think so, I was hoping we'd unify the code paths a bit more but that make sense.